addons/website/models/website.py

```diff
@@ -183,10 +184,19 @@ def _bootstrap_homepage(self):
         </t>''' % (self.id)
         standard_homepage.with_context(website_id=self.id).arch_db = new_homepage_view
 
-        self.homepage_id = self.env['website.page'].search([('website_id', '=', self.id),
-                                                            ('key', '=', standard_homepage.key)])
+        homepage_page = Page.search([
+            ('website_id', '=', self.id),
+            ('key', '=', standard_homepage.key),
+        ])
+        if not homepage_page:
+            homepage_page = Page.create({
+                'website_published': True,
+                'url': '/',
+                'view_id': self.with_context(website_id=self.id).viewref('website.homepage').id,
```

Comment:
if you delete the homepage page, it delete the view... so what will returned by self.with_context(website_id=self.id).viewref('website.homepage').id?

Reply:
Yes I was unclear when writting commit, the problematic case is when the view has no page_id anymore and the many2one link is broken.

The code was already handling the case where the view homepage was removed.
Then the rest of the code was assuming if the view exists and was COW, the website.page did also exists, which is not true all the time.

I'll update the commit message then we can merge or cancel this one, but we already had a few tickets for this one.